eZ/Bundle/EzPublishCoreBundle/Resources/config/papi.yml

```diff
@@ -54,11 +56,18 @@ services:
         factory_method: buildStorageEngine
         public: false
 
+    ezpublish.api.search_engine.factory:
+        class: %ezpublish.api.search_engine.factory.class%
+        arguments:
+            - @ezpublish.api.repository_configuration_provider
+
     ezpublish.spi.search:
         alias: ezpublish.spi.search_engine
 
     ezpublish.spi.search_engine:
-        alias: ezpublish.spi.search.legacy
+        class: %ezpublish.api.search_engine.class%
+        factory_service: ezpublish.api.search_engine.factory
+        factory_method: buildSearchEngine
```

Comment:
Fixed in https://github.com/ezsystems/ezpublish-kernel/commit/b7fbbef6477ef26e2d238e7afb5c336bdc33fc0f